### ISSUE TITLE
Grammer causes exception in vscode-textmate

### ISSUE THREAD

**Issue by matter123** [jeff-hykin/cpp-textmate-grammar#40]
The grammar causes visual studio code to have a type error in the form of ```[renderer1] [error] Cannot create property 'id' on string 't': TypeError: Cannot create property 'id' on string 't'``` `git-bisect` has determined that the offending commit is 216543b470c381541a7bc9d0c60a85b5d513c0c4. The issue is found in VSCode 1.32 and insiders as well as the standalone https://github.com/Microsoft/vscode-textmate/ package. The type error seems to have little if any impact on the actual application of scopes, It does however seem to occasionally cause the theme to not be applied.

<img width="657" alt="Screen Shot 2019-03-20 at 1 44 25 PM" src="https://user-images.githubusercontent.com/714007/54717840-57728f00-4b16-11e9-818e-8f341a680ce7.png">

example:
```
namespace test {
	class test {
		test(){};
	};
} // namespace test
```

**Comment by matter123** [comment]
@jeff-hykin what is `injections: "testing"` supposed to do? I can't find any reference to the grammar `testing` in the repository or on the web and `injections` doesn't seem to be the documented way of injecting a grammar.

**Comment by matter123** [comment]
`injections` is a internal field in the type `IRawGrammar` when `vscode-textmate` parses the grammar, the value supplied by the grammar replaces `injections`. This results in `injections` having the wrong type. Removing `injections` from the grammar definition fixes the issue.

**Comment by jeff-hykin** [comment]
Sorry it auto closed this issue on the merge. I added that field since some of the other TextMate languages had that field, and I wanted the 'readable_grammar.rb' to work for any field.

Here's an example of them in other languages.
<img width="338" alt="Screen Shot 2019-03-21 at 4 51 01 PM" src="https://user-images.githubusercontent.com/17692058/54788839-924ff200-4bfe-11e9-953b-c1e78c0552a8.png">

That said, it is completely unneeded for C++ and C.

**Comment by matter123** [comment]
Well if kept, the output value needs to be a json object as can be seen in html.tmlanguage.json.

**Comment by matter123** [comment]
Fixed with 1.6.6
